### PR TITLE
Added coin change problem

### LINKED ISSUE #126: Article for Coin Change Problem
## 💥 Proposal

I would like to add an article for Coin Change Problem. The basic layout for the article will be:

What is Coin Change Problem?
Explanation for Coin Change Problem.
Algorithmic Steps
Code (Use C++/Java and Python; At least two languages should be present and pick up code from NeoAlgo with apt credits to author)
Sample I/O
Time & Space Complexity
This article should be between 500-700 words in length. You should use traditional Markdown to prepare the content and the use of external images is prohibited. You can however create your own images. Run a proper Grammarly Lint check before submitting the PR.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

### PR DESCRIPTION
**Pull request by DEEPALI-SUK**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description

Added the article for Coin Change Problem. The basic layout for the article is:
What is the Coin Change Problem?
Explanation for Coin Change Problem
Algorithmic Steps
Example
Code in Java and Python
Sample I/O
Time & Space Complexity
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes  #126

### CODE REVIEW

**Comment by DEEPALI-SUK** [comment]
@HarshCasper Kindly, review my PR. I have done the necessary changes suggested by you.

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+- Make a 2-d array, say `dp[n+1][amount+1]`.
+- Assign all the columns of the `0th` row of the 2d-array to a maximum value: `dp[0][i]= INT_MAX`.
+- Assign all the rows of the `0th` column of the 2-d array as 0, that is, `dp[i][0]=0`.
+- For the `1st` row, if the column indices value is divisible by the first coin, write the quotient in that cell, else write the max value.
+- For row>1, iterate till column index value is less than the ith coin value (i.e for all column indices, j,such that, j<coins[i-1] )and assign `dp[i][j]` to the previous upper row value, that is, `dp[i][j]=dp[i-1][j]`.
+- Now we have two options for all column indices, j, such that, j>=coins[i-1], whether to include ith coin or not.
+- Case1- If we don't consider the ith coin, then the cell value is the previous row cell value that is, `dp[i][j]=dp[i-1][j]`.
+- Case2- If we consider the ith coin, then the cell value is `1+ dp[i][j-coins[i-1]]` cell value.
+- The final value of the cell is the minimum of both the cases,that is, `dp[i][j]= min(dp[i-1][j], 1+dp[i][j-coins[i-1]])`.
+- The final answer will be `dp[n][amount]`.
```
Comment (HarshCasper) [inline docs/dynamic-programming/coin-change-problem.mdx:25]: It looks too cluttered. Any ways to make it easy and compact to understand?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/dynamic-programming/coin-change-problem.mdx (hunk 1 above)
Comment (DEEPALI-SUK) [reply docs/dynamic-programming/coin-change-problem.mdx:39]: Sorry, I have already simplified this algorithm, and I am not able to simplify it more. 

> It looks too cluttered. Any ways to make it easy and compact to understand?

**Comment by DEEPALI-SUK** [comment]
@HarshCasper Kindly, review my PR. I have done rest of the changes suggested by you.

**Comment by DEEPALI-SUK** [comment]
@iamrajiv @atarax665 Kindly, review my PR, I have done all the suggested changes.

File: docs/dynamic-programming/coin-change-problem.mdx [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+}>
+
+<TabItem value="java">
+
+```java
+import java.util.Scanner;
+
+class CoinChangeUsingDp {
+
+    public static int coinChange(int[] coins, int amount) {
```
Comment (bhav09) [inline docs/dynamic-programming/coin-change-problem.mdx:61]: Can you please comment this section of your code till your main function.

**Review by bhav09 — CHANGES_REQUESTED** [review]

File: docs/dynamic-programming/coin-change-problem.mdx (hunk 2 above)
Comment (DEEPALI-SUK) [reply docs/dynamic-programming/coin-change-problem.mdx:61]: image


Please could you tell me whether this is correct?  Or please could you specify the line numbers which I have to comment out.
Thank you

Comment (DEEPALI-SUK) [reply docs/dynamic-programming/coin-change-problem.mdx:61]: I have added the comments now.

**Comment by DEEPALI-SUK** [comment]
@bhav09 Kindly, merge my PR.

**Pull request merged**